docs/proposals/promote-dependencies-automatically/README.md

```diff
@@ -0,0 +1,196 @@
+3. Custom Interpretation
+   1. Lua Configuration-based Interpretation:
+      1. Configure resource interpretation rules for dependency parsing by specifying Lua scripts in the [ResourceInterpreterCustomization](https://github.com/karmada-io/karmada/blob/84b971a501ba82c53a5ad455c2fe84d842cd7d4e/pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go#L17) resource.
+   2. Webhook-based Interpretation:
+      1. Users can design Resource Interpreter Webhooks for specific custom resources to handle Interpreter Operations requests and perform dependency parsing. The process involves receiving these request operations and processing them accordingly.
+         The specific process is as follows:
+
+![webhook-interpret](statics/webhook-interpret.png)
+
+***Note:*** 
+Using the webhook-based interpretation method may pose difficulties when making requests to the webhook interpret service. In such cases, we need to perform port forwarding to invoke the service within the cluster.
+
+### Conflict and Failure Resolution
+
+#### Conflict Resolution
+
+When promoting a resource, if the resource to be promoted or its dependent resources have already been promoted to the Karmada control plane, the promotion operation will abort the promotion of the resource or its dependent resources. The user needs to delete the existing resource in the Karmada control plane before promoting it again.
+
+#### Failure Resolution
+
+When promoting a resource, if the resource to be promoted and its dependent resources fail to be promoted to the Karmada control plane(validation failed or other), the promotion operation will fail and will not promote any resources to the Karmada control plane.
+
+### Example
+
+Assuming we deploy a workload CR named `myapp` and its referenced ConfigMap called `my-config` in the member cluster member1.
+
+```yaml
+apiVersion: apps/v1
+kind: Workload
+metadata:
+  name: myapp
+  labels:
+    app: myapp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: myapp
+  template:
+    metadata:
+      labels:
+        app: myapp
+    spec:
+      containers:
+        - image: nginx
+          name: nginx
+          volumeMounts:
+            - name: configmap
+              mountPath: "/configmap"
+      volumes:
+        - name: configmap
+          configMap:
+            name: my-config
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  nginx.properties: |
+    proxy-connect-timeout: "10s"
+    proxy-read-timeout: "10s"
+    client-max-body-size: "2m"
+```
+After executing the `karmadactl promote workload myapp -c member1 -d=true` command, we can obtain the Resource Binding and Work object corresponding to the workload `myapp` and ConfigMap `my-config` in the Karmada control plane. This indicates that both `myapp` and its dependent resource `my-config` have been successfully promoted to the Karmada control plane.
```

Comment:
Do we have similar information output like the following? To prompt users about which resources have been `promoted`.

```
# kubectl create -f test.yaml
deployment.apps/myapp created   
configmap/my-config created
```
for example:

```
`karmadactl promote workload myapp -c member1 -d=true` 
workload.apps/myapp promoted   
configmap/my-config promoted
```


other LGTM

Reply:
Yes, like this:
![image](https://github.com/karmada-io/karmada/assets/56665618/b460ca24-0d32-44c2-bcbd-844414bc8d0c)

Reply:
Fantastic!